test/main-spec.js

```diff
@@ -12,185 +12,451 @@
 
```

Comment:
The second idea seems to be clearer and better readable. I would call the `spec-helper` something like `main-suite.js`. And I've tried: It is not necessary to pass `describe` and `it` into the function. They are globals. I think, everything can go into the `main-suite.js` except 
```js
var promisedHandlebars = require('../')
var Handlebars = promisedHandlebars(require('handlebars'), {Promise: require('q').Promise})
require('./default-suite')(Handlebars, 'Q')
```

The part where you check for the thrown exception should be part of the `default-promise-spec.js`. It shouldn't call the default suite, if `global.Promise` does not exist.